Sources/Subs-Membergroups.php

```diff
@@ -830,4 +830,70 @@ function list_getMembergroups($start, $items_per_page, $sort, $membergroup_type)
 	return $groups;
 }
 
+/**
+ * Retrieves a list of membergroups with the given permissions.
+ *
+ * @param array $group_permissions
+ * @param array $board_permissions
+ * @param int   $profile_id
+ *
+ * @return array An array containing two arrays - 'allowed', which has which groups are allowed to do it and 'denied' which has the groups that are denied
+ */
+function getGroupsWithPermissions(array $group_permissions = array(), array $board_permissions = array(), $profile_id = 1)
+{
+	global $smcFunc;
+
+	$member_groups = array();
+	if (!empty($group_permissions))
```

Comment:
You can avoid unneeded nesting by checking for empty($group_permissions) and return early

Reply:
No